backends/dpdk/dpdkArch.cpp

```diff
@@ -3035,4 +3035,169 @@ std::vector<std::pair<cstring, const IR::Type *>>
 cstring DpdkAddPseudoHeaderDecl::pseudoHeaderInstanceName;
 cstring DpdkAddPseudoHeaderDecl::pseudoHeaderTypeName;
 
+void InsertReqDeclForIPSec::uniqueNames() {
+    // Following names are reserved for DPDK IPSec implementation
+    // make sure no such decls exist with that name
+    registerInstanceNames[0] = "ipsec_port_out_inbound";
+    registerInstanceNames[1] = "ipsec_port_out_outbound";
+    registerInstanceNames[2] = "ipsec_port_in_inbound";
+    registerInstanceNames[3] = "ipsec_port_in_outbound";
+    cstring name = "";
+    for (unsigned i = 0; i < 4; i++) {
+        name = refMap->newName(registerInstanceNames[i]);
+        if (name != registerInstanceNames[i])
+            ::error("%1% name is reserved for DPDK IPSec port register", registerInstanceNames[i]);
+    }
+}
+
+// create and add register declaration instance to program
+IR::Declaration_Instance *InsertReqDeclForIPSec::addRegDeclInstance(cstring instanceName) {
```

Comment:
can we make it also generic and put in utility, all configurable part passed as param?

Reply:
You mean to add in DpdkUtils to be used by other passes as well?